plutus-core/test/Evaluation/DynamicBuiltins/Definition.hs

```diff
@@ -302,31 +139,15 @@ argument when it's a function, for another example).
 test_IdRank2 :: TestTree
 test_IdRank2 =
     testCase "IdRank2" $ do
-        let tyAct = typeOfBuiltinFunction @DefaultUni IdRank2
-            tyExp = let f = TyName . Name "f" $ Unique 0
-                        a = TyName . Name "a" $ Unique 1
-                        allAfA = TyForall () a (Type ()) . TyApp () (TyVar () f) $ TyVar () a
-                    in TyForall () f (KindArrow () (Type ()) $ Type ()) $ TyFun () allAfA allAfA
-            res = mkConstant @Integer @DefaultUni () 0
+        let res = mkConstant @Integer @DefaultUni () 0
             integer = mkTyBuiltin @Integer ()
             -- sum (idRank2 {list} nil {integer})
             term
                 = Apply () (mapFun Left Plc.sum)
                 . TyInst () (Apply () (TyInst () (Builtin () $ Right IdRank2) Plc.listTy) Plc.nil)
                 $ integer
-        tyAct @?= tyExp
         typecheckEvaluateCkNoEmit defBuiltinsRuntimeExt term @?= Right (EvaluationSuccess res)
 
--- | Test that the type of PLC @Absurd@ is inferred correctly.
-test_Absurd :: TestTree
-test_Absurd =
-    testCase "Absurd" $ do
-        let tyAct = typeOfBuiltinFunction @DefaultUni Absurd
-            tyExp = let a = TyName . Name "a" $ Unique 0
-                        tyForallA = TyForall () a (Type ())
-                    in tyForallA . TyFun () (tyForallA $ TyVar () a) $ TyVar () a
-        tyAct @?= tyExp
```

Comment:
Now that we have golden tests, we don't need to manually check that types are inferred correctly. I decided to keep one, though (`idList`) as the golden file for it is unreadable.

Reply:
Doesn't that depend on printing the types correctly?  If we were unlucky there might be a bug in the printer that causes two different types to be printed in the same way, and we wouldn't spot the problem...

Reply:
Yes, but it's also possible to construct the AST incorrectly and not notice the problem, there are always risks involved. And I'd argue that it's more likely to mess up the AST than the pretty-printer. Plus, those ASTs are unreadable while PLC types are somewhat readable in simple cases, so they can be visually inspected. But the main reason for using golden tests is that creating AST is tedious and I want to avoid that when possible. I could keep the already existing test cases though instead of deleting them if people think it's worth it. @michaelpj ?